int64 num_point_1
int64 num_point_2